nyxtools/flyer.py

```diff
@@ -0,0 +1,177 @@
+        #     "spec": "AD-EIGER-MX",
+        #     ...: ...,
+        # }
+        # datum = {
+        #     "datum_id": "a",
+        #     "datum_kwargs": {"data_key": "data"},  # this goes to __call__
+        #     "resource": "RES",
+        #     ...: ...,
+        # }
+        # datum = {
+        #     "datum_id": "b",
+        #     "datum_kwargs": {"data_key": "omega"},
+        #     "resource": "RES",
+        #     ...: ...,
+        # }
+
+        # event = {...: ..., "data": {"detector_img": "a", "omega": "b"}}
+
+        for data_key in self._datum_ids.keys():
+            datum_id = f"{resource_uid}/{data_key}"
+            self._datum_ids[data_key] = datum_id
+            datum = {
+                "resource": resource_uid,
+                "datum_id": datum_id,
+                "datum_kwargs": {"data_key": data_key},
+            }
+            asset_docs_cache.append(("datum", datum))
+        return tuple(asset_docs_cache)
+
+    def _extract_metadata(self, field="omega"):
+        with fabio.open(self._first_file, "r") as cbf:
+            return cbf.pilatus_headers("omega")
+
+    def detector_arm(self, **kwargs):
+        start = kwargs["angle_start"]
+        width = kwargs["img_width"]
+        num_images = kwargs["num_images"]
+        exposure_per_image = kwargs["exposure_period_per_image"]
+        file_prefix = kwargs["file_prefix"]
+        data_directory_name = kwargs["data_directory_name"]
+        x_beam = kwargs["x_beam"]
+        y_beam = kwargs["y_beam"]
+        wavelength = kwargs["wavelength"]
+        det_distance_m = kwargs["det_distance_m"]
+
+        self.detector.cam.save_files.put(1, wait=True)
+        self.detector.cam.file_owner.put(getpass.getuser(), wait=True)
+        self.detector.cam.file_owner_grp.put(grp.getgrgid(os.getgid())[0], wait=True)
+        self.detector.cam.file_perms.put(420, wait=True)
+        file_prefix_minus_directory = str(file_prefix)
+        file_prefix_minus_directory = file_prefix_minus_directory.split("/")[-1]
+
+        self.detector.cam.acquire_time.put(exposure_per_image, wait=True)
+        self.detector.cam.acquire_period.put(exposure_per_image, wait=True)
+        self.detector.cam.num_images.put(num_images, wait=True)
+        self.detector.cam.file_path.put(data_directory_name, wait=True)
+        self.detector.cam.file_name.put(file_prefix_minus_directory, wait=True)
+
+        # originally from detector_set_fileheader
+        self.detector.cam.beam_center_x.put(x_beam, wait=True)
+        self.detector.cam.beam_center_y.put(y_beam, wait=True)
+        self.detector.cam.omega_incr.put(width, wait=True)
+        self.detector.cam.omega_start.put(start, wait=True)
+        self.detector.cam.wavelength.put(wavelength, wait=True)
+        self.detector.cam.det_distance.put(det_distance_m, wait=True)
+
+        start_arm = ttime.time()
+        self.detector.cam.acquire.put(1, wait=True)
+        logger.info(f"arm time = {ttime.time() - start_arm}")
```

Comment:
It's better to use https://docs.python.org/3/library/time.html#time.monotonic:
```suggestion
        start_arm = ttime.monotonic()
        self.detector.cam.acquire.put(1, wait=True)
        logger.info(f"arm time = {ttime.monotonic() - start_arm}")
```

Reply:
agree with this - will commit the suggestion